### ISSUE TITLE
Support SQL `IS` binary operator

### ISSUE THREAD

**Issue by FPtje** [bitemyapp/esqueleto#358]
I found myself manually creating the following definition:

```haskell
is :: Esq.SqlExpr (Esq.Value a) -> Esq.SqlExpr (Esq.Value a) -> Esq.SqlExpr (Esq.Value Bool)
is = EsqInternal.unsafeSqlBinOp "IS"
```

Equivalently, one can imagine the following operator:

```haskell
is_not :: Esq.SqlExpr (Esq.Value a) -> Esq.SqlExpr (Esq.Value a) -> Esq.SqlExpr (Esq.Value Bool)
is_not = EsqInternal.unsafeSqlBinOp "IS NOT"
```
(although this one can be done with the `not_ (is ...)` function)

The `IS` binary operator is most commonly seen in `IS NULL` and `IS NOT NULL`. esqueleto indeed has this in the `isNothing` function, but they can also be used for other values, e.g. `foo IS true`. The reason to use `IS` over `=` here is because they have different behaviors regarding NULL.

Here are the truth tables for  `x = y` and `x IS y` for booleans:

| `x = y` | `true`  | `false` | `NULL` |
| ------- | ------- | ------- | ------ |
| `true`  | `true`  | `false` | `NULL` |
| `false` | `false` | `true`  | `NULL` |
| `NULL`  | `NULL`  | `NULL`  | `NULL` |

| `x IS y` | `true`  | `false` | `NULL` |
| -------- | ------- | ------- | ------ |
| `true`   | `true`  | `false` | `false` |
| `false`  | `false` | `true`  | `false` |
| `NULL`   | `false`  | `false`  | `true` |

I didn't make a pull request because this function might be up to debate. Arguably, one might say that with a correct Persistent table definition, you statically know whether a column is NULL or not, so the `IS` operator may not technically be needed. On the other hand, there might be other expressions where `NULL` sneaks in to give funny behavior. What do you think?

---

Also, apologies if this issue is duplicate. It's kind of hard to search for a two-letter operator :sweat_smile:

**Comment by belevy** [comment]
looking this up, is can only be used for trivalued booleans (true, false, null)

**Comment by parsonsmatt** [comment]
So the proper type here is more like `SqlExpr (Value (Maybe Bool)) -> SqlExpr (Value (Maybe Bool)) -> SqlExpr (Value (Maybe Bool))`?

**Comment by Vlix** [comment]
Should be `is :: SqlExpr (Value (Maybe Bool)) -> SqlExpr (Value (Maybe Bool)) -> SqlExpr (Value Bool)`, no?

**Comment by parsonsmatt** [comment]
Yes, good catch- thanks!

**Comment by FPtje** [comment]
I hadn't realized that the use is limited to true/false/null. The signature looks good to me 👍

**Comment by Vlix** [comment]
Yeah, it's limited to `Bool` and `Maybe a`, so I'd personally probably want the API to be more like:

```hs
-- Boolean operator
is :: SqlExpr (Value (Maybe Bool)) -> SqlExpr (Value (Maybe Bool)) -> SqlExpr (Value Bool)
-- Synonym for `isNothing`
isNull :: SqlExpr (Value (Maybe a)) -> SqlExpr (Value Bool)
-- Synonym for `not_ . isNothing`
isNotNull :: SqlExpr (Value (Maybe a)) -> SqlExpr (Value Bool)
```
